save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/contests/ContestInfoMenu.kt

```diff
@@ -4,51 +4,130 @@ package com.saveourtool.save.frontend.components.basic.contests
+            responseHandler = ::noopResponseHandler,
+        )
+            .unsafeMap {
+                it.decodeFromJsonString<TestFilesContent>()
+            }
+        setPublicTest(publicTestDto)
+    }()
+
     div {
-        attrs.className = ClassName("d-flex justify-content-around")
+        className = ClassName("d-flex justify-content-around mb-3")
         div {
-            attrs.className = ClassName("col-4")
+            className = ClassName("col-5")
             div {
-                attrs.className = ClassName("text-xs text-center font-weight-bold text-primary text-uppercase mb-3")
+                className = ClassName("text-xs text-center font-weight-bold text-primary text-uppercase mb-3")
                 +"Description"
             }
             div {
-                attrs.className = ClassName("text-center")
-                child(cardComponent(hasBg = true) { +(props.contest?.description ?: "") })
+                className = ClassName("text-center")
+                card {
+                    child(reactMarkdown(jso {
+                        this.children = contest?.description ?: "No description provided **yet**"
+                    }))
+                }
             }
         }
         div {
-            attrs.className = ClassName("col-4")
+            className = ClassName("col-5")
             div {
-                attrs.className = ClassName("text-xs text-center font-weight-bold text-primary text-uppercase mb-3")
+                className = ClassName("text-xs text-center font-weight-bold text-primary text-uppercase mb-3")
                 +"Public tests"
             }
             div {
-                attrs.className = ClassName("text-center")
-                child(cardComponent(hasBg = true, isBordered = true) {
-                    +"PUBLIC TESTS WILL BE HERE"
-                })
+                if (publicTest.testLines.isEmpty()) {
+                    div {
+                        className = ClassName("text-center")
+                        +"Public tests are not provided"
+                    }
+                } else {
+                    card {
+                        div {
+                            className = ClassName("ml-2 mr-2")
+                            div {
+                                className = ClassName("mt-3 mb-3")
+                                displayTestLines("Test", publicTest.testLines)
+                            }
+                            publicTest.expectedLines?.let {
+                                div {
+                                    className = ClassName("mt-3 mb-2")
+                                    displayTestLines("Expected", it)
+                                }
+                            }
+                        }
+                    }
+                }
             }
         }
     }
 }
+
+private fun wrapTestLines(testLines: List<String>, language: String?) = "```${language ?: ""}\n${testLines.joinToString("\n")}\n```"
```

Comment:
```suggestion
private fun wrapTestLines(testLines: List<String>, language: String?) = """
    |```${language ?: ""}
    |${testLines.joinToString("\n")}
    |```""".trimMargin()
```
I think it looks more clear

Reply:
It does, thanks